src/ale.cpp

```diff
@@ -266,4 +254,115 @@ namespace ale {
    return result;
  }
 
+ std::string getPyTraceback() {
+    PyObject* err = PyErr_Occurred();
+    if (err != NULL) {
+        PyObject *ptype, *pvalue, *ptraceback;
+        PyObject *pystr, *module_name, *pyth_module, *pyth_func;
+        char *str;
+        char *full_backtrace;
+        char *error_description;
+
+        PyErr_Fetch(&ptype, &pvalue, &ptraceback);
+        pystr = PyObject_Str(pvalue);
+        str = PyBytes_AS_STRING(PyUnicode_AsUTF8String(pystr));
+        error_description = strdup(str);
+
+        // See if we can get a full traceback
+        module_name = PyUnicode_FromString("traceback");
+        pyth_module = PyImport_Import(module_name);
+        Py_DECREF(module_name);
+
+        if (pyth_module == NULL) {
+            throw runtime_error("getPyTraceback - Failed to import Python traceback Library");
+        }
+
+        pyth_func = PyObject_GetAttrString(pyth_module, "format_exception");
+        PyObject *pyth_val;
+        pyth_val = PyObject_CallFunctionObjArgs(pyth_func, ptype, pvalue, ptraceback, NULL);
+
+        pystr = PyObject_Str(pyth_val);
+        str = PyBytes_AS_STRING(PyUnicode_AsUTF8String(pystr));
+        full_backtrace = strdup(str);
+        Py_DECREF(pyth_val);
+
+        std::string join_cmd = "trace = ''.join(list(" + std::string(full_backtrace) + "))";
+        PyRun_SimpleString(join_cmd.c_str());
+
+        PyObject *evalModule = PyImport_AddModule( (char*)"__main__" );
+        PyObject *evalDict = PyModule_GetDict( evalModule );
+        PyObject *evalVal = PyDict_GetItemString( evalDict, "trace" );
+        full_backtrace = PyBytes_AS_STRING(PyUnicode_AsUTF8String(evalVal));
+
+        return std::string(error_description) + "\n" + std::string(full_backtrace);
+    }
+    
+    // no traceback to return
+    return "";
+ }
+
+ std::string load(std::string filename) {
+     static bool first_run = true;
+     if(first_run) {
+         // Initialize the Python interpreter but only once.
+         first_run = !first_run;
+         Py_Initialize();
+         atexit(Py_Finalize);
```

Comment:
`Py_Initialize` and `Py_Finalize` should only be called once in a programs lifetime, this should ensure that this is enforced.